spring-cloud-data-core/src/main/java/org/springframework/cloud/data/core/dsl/TaskDslParser.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.data.core.dsl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Michael Minella
+ */
+public class TaskDslParser {
+
+	private String expressionString;
+
+	private List<Token> tokenStream;
+
+	private int tokenStreamLength;
+
+	private int tokenStreamPointer; // Current location in the token stream when
+
+	// processing tokens
+
+	private int lastGoodPoint;
+
+	/**
+	 * Parse a task definition without supplying the task name up front.
+	 * The task name may be embedded in the definition.
+	 * For example: <code>myTask = filejdbc</code>
+	 *
+	 * @return the AST for the parsed stream
+	 */
+	public ModuleNode parse(String stream) {
```

Comment:
So a couple thoughts here:

@ilayaperumalg: I left the variable names as stream because in the context they are used, it's not really a stream as in an XD stream...it's a stream of characters being parsed.

@markfisher: I can definitely do that.  This was actually just a quick cut of the parsing stuff because I needed something to be able to get to the launching piece (the more interesting piece).  I had it all in one commit but I felt it was too large for a single PR.  Would you rather I retract this and go for that refactoring now or get the basics for launching contributed and revisit the parsing logic afterwards?